python/pylibcugraph/pylibcugraph/heterogeneous_biased_neighbor_sample.pyx

```diff
@@ -0,0 +1,465 @@
+
+    # Have the SamplingResult instance assume ownership of the result data.
+    result = SamplingResult()
+    result.set_ptr(result_ptr)
+
+    # Get cupy "views" of the individual arrays to return. These each increment
+    # the refcount on the SamplingResult instance which will keep the data alive
+    # until all references are removed and the GC runs.
+    # TODO Return everything that isn't null in release 23.12.
+    if with_edge_properties:
+        cupy_majors = result.get_majors()
+        cupy_major_offsets = result.get_major_offsets()
+        cupy_minors = result.get_minors()
+        cupy_edge_weights = result.get_edge_weights()
+        cupy_edge_ids = result.get_edge_ids()
+        cupy_edge_types = result.get_edge_types()
+        cupy_batch_ids = result.get_batch_ids()
+        cupy_label_hop_offsets = result.get_label_hop_offsets()
+
+        if renumber:
+            cupy_renumber_map = result.get_renumber_map()
+            cupy_renumber_map_offsets = result.get_renumber_map_offsets()
+            # TODO drop the placeholder for hop ids in release 23.12
+            if return_dict:
+                return {
+                    'major_offsets': cupy_major_offsets,
+                    'majors': cupy_majors,
+                    'minors': cupy_minors,
+                    'weight': cupy_edge_weights,
+                    'edge_id': cupy_edge_ids,
+                    'edge_type': cupy_edge_types,
+                    'batch_id': cupy_batch_ids,
+                    'label_hop_offsets': cupy_label_hop_offsets,
+                    'hop_id': None,
+                    'renumber_map': cupy_renumber_map,
+                    'renumber_map_offsets': cupy_renumber_map_offsets
+                }
+            else:
+                cupy_majors = cupy_major_offsets if cupy_majors is None else cupy_majors
+                return (cupy_majors, cupy_minors, cupy_edge_weights, cupy_edge_ids, cupy_edge_types, cupy_batch_ids, cupy_label_hop_offsets, None, cupy_renumber_map, cupy_renumber_map_offsets)
+        else:
+            cupy_hop_ids = result.get_hop_ids() # FIXME remove this
+            if return_dict:
+                return {
+                    'major_offsets': cupy_major_offsets,
+                    'majors': cupy_majors,
+                    'minors': cupy_minors,
+                    'weight': cupy_edge_weights,
+                    'edge_id': cupy_edge_ids,
+                    'edge_type': cupy_edge_types,
+                    'batch_id': cupy_batch_ids,
+                    'label_hop_offsets': cupy_label_hop_offsets,
+                    'hop_id': cupy_hop_ids,
+                }
+            else:
+                cupy_majors = cupy_major_offsets if cupy_majors is None else cupy_majors
+                return (cupy_majors, cupy_minors, cupy_edge_weights, cupy_edge_ids, cupy_edge_types, cupy_batch_ids, cupy_label_hop_offsets, cupy_hop_ids)
+
+    else:
```

Comment:
I don't think we should move over deprecated functionality to this new API.  I don't think we should support `with_edge_properties`.  When I implemented biased sampling in PLC, I didn't add any deprecated arguments.

Reply:
Got it. I removed `with_edge_properties` and I am only returning the results as a dictionary.